README reworked (#1444)
retrospect-addon/plugin.video.retrospect: README.md

[![Python](https://img.shields.io/badge/python-2.7%20%7C%203.6-blue?logo=python)](https://kodi.tv/article/attention-addon-developers-migration-python-3)
[![Transifex](https://img.shields.io/badge/transifex-en|nl|sv|no|fi-blueviolet)](https://www.transifex.com/retrospect/public/)

This repository holds the main code for Retrospect. For more info on reporting bugs, please visit https://github.com/retrospect-addon/plugin.video.retrospect or https://www.rieter.net/content/.

# Introduction #

> Retrospect (Previously XBMC Online TV) is a Kodi video add-on which allows you to watch streams of 

Author: ```suggestion
Retrospect is a Kodi video add-on allowing you to watch streams from 
```

> a number of gratis publicly available online TV stream sites.

Reviewer: I would keep "free and publicly", as "gratis" seems so old fashion.

Author: Changed it around to be less awkward. The problem with "free" is that it may be confused for "Free Software TV streaming sites".

Reviewer 2: Didn't see this, sorry. As I commented below, I suggest:
"allows you to watch a number of freely available online TV streaming sites"

Gratis is indeed archaic. Why "free" would cause that confusion I don't understand, @comradekingu.

Also, remove "(Previously XBMC Online TV)".

> a number of gratis publicly available online TV stream sites.

Author: ```suggestion
a number of publicly available and gratis online TV stream sites.
```

> a number of gratis publicly available online TV stream sites.

Author: ```suggestion
a number of gratis publicly available online TV streaming sites.
```

Reviewer 2: Why do you want to use gratis instead of free? Suggestion:
"allows you to watch a number of freely available online TV streaming sites"

Also, remove "(Previously XBMC Online TV)".

> a number of publicly available and gratis online TV stream sites.

Author: ```suggestion
a number of publicly available and gratis online TV streaming sites.
```

> a number of publicly available and gratis online TV stream sites.

Author: ```suggestion
a number of publicly available and gratis online TV and video-streaming sites.
```

It all started with the streams from the Dutch site [www.uitzendinggemist.nl](https://npostart.nl).
The add-on got its initial name from this site: ‘_Uitzending gemist_’ meaning 
‘_missed broadcast_’ in Dutch.

Discuss the add-on can be done at [the Kodi forums thread](https://forum.kodi.tv/showthread.php?tid=25522),
or at [the Tweakers.net thread](https://gathering.tweakers.net/forum/list_messages/1643928/last) (Dutch only).
Issues regarding this script can be submitted at our issue tracker at GitHub.

# Installing Retrospect #
There are two ways of installing Retrospect, (depending on the Kodi version you are using).

### Kodi Leia and later
Starting from Kodi Leia (v18), you can install Retrospect from the official Kodi add-on repository. Use the search function in the add-ons section to find `Retrospect` and install it. Detailed info can be found in the [Retrospect wiki](https://github.com/retrospect-addon/plugin.video.retrospect/wiki/Installation).

> **Note:** Make sure the _Auto-Update_ option for Retrospect is turned on to automagically receive new updates.

Reviewer: It really is `enabled`, that is how it is called for almost all Kodi (and add-on) settings.

Author: Unless it can be turned on, and disabled at the same time, semantically you are adding complexity. Also makes it sound less welcoming. I put in "Turn on/off" for Kodi stuff when I can.

![alt text](./resources/media/retrospect01.jpg "The Retrospect information screen")

### Kodi Krypton    
If you are running Kodi Krypton (v17) the only way to install Retrospect is to install it from a ZIP file. Follow these steps (keep in mind they might slightly differ depending on your skin):
1. Download the latest Retrospect ZIP file from here: [https://github.com/retrospect-addon/plugin.video.retrospect/releases](https://github.com/retrospect-addon/plugin.video.retrospect/releases)
1. Transfer the file to your Kodi system (_optional_).
1. Use `Install from ZIP file` in Kodi to install Retrospect.

More detailed info can be found in the [Retrospect wiki](https://github.com/retrospect-addon/plugin.video.retrospect/wiki/Installation).

**Note:** Since Retrospect isn't available in the official Kodi add-on repository for Kodi Krypton (v17), it will not automagically update.
Retrospect will notify you of new versions. You will then have to manually update Retrospect using the steps above.

# Contributing #

> You can help developing Retrospect via our [GitHub](https://github.com/retrospect-addon/plugin.video.retrospect) page, and/or help translating Retrospect via [Hosted Weblate](https://hosted.weblate.org/projects/retrospect/#components) (A big thanks to Weblate for their services).

Reviewer: I don't use Weblate (yet), so I don't know why you added it?

Author: To smooth out the process of leaving that platform.

Reviewer: Ok, but then two things need to happen: Weblate needs to aprove the "Free" version and I would like to move to GPL3 or APGL3 to get rid of the CC license.

Author: You can click to ask for approval any time, and changing the license is possible in the Weblate settings.

# Troubleshooting #

### Playing Widevine DRM content ###
Starting from Kodi Leia (v18), playback of DRM protected streams is supported using the add-on `InputStream Adaptive`, which is automatically installed along with Retrospect.

In order to play Widevine DRM files you will also need to have the `Google Widevine libraries` installed. Android based devices have this installed as a native component, for Windows and Linux you have to install them first:

The _Easy way_:

1. Open the Retrospect add-on settings.

> 1. Under the "General" tab, make sure `Use Kodi InputStream Adaptive add-on when possible` is on.

Reviewer: Again, this really should be `enabled`

1. From that recovery image, you will need the Widevine files located in /opt/google/chrome/libwidevinecdm*.so.
1. These files need to be copied to the `<kodi-profile>/cdm` folder.

_Note: Keep in mind you might need to try multiple recovery images before you find a working one._ 

# Acknowledgement #
The first idea for Retrospect/XBMC Online TV/XOT-Uzg came from a script by BaKMaN.

# Copyrights and Licenses #

> Retrospect is licensed _Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International_, see: [LICENSE.md](./LICENSE.md)

Reviewer: It really should be `licensed under a`. Then even use that on their own website (https://creativecommons.org/licenses/)

Author: They use Google Analytics on their site. Just about everything about CC happens to be sub-par.

Reviewer: This will change to (A)GPL3 soon.

Author: Great. Will update it once that happens.

> The official add-on package for Retrospect is not distributed via other repositories than the official Kodi add-on repository.

Reviewer: This actually means "you are not allowed to distribute it otherwise than the Kodi repo"

Author: No, it just means those aren't official ones, which holds true.

Reviewer 2: Nope, "may not" means what Bas says. However, I would suggest just "Retrospect" as it's self-evident it's the package.

Author: The original or the new version? It is permitted by the license, and then it becomes a question of what is official and what is not.

### Disclaimer ###
Retrospect or Rieter.net are not connected to or in any other way affiliated with Kodi, Team Kodi or the XBMC Foundation. Furthermore, any software, addons, or products offered by Retrospect or Rieter.net will only receive support in its [Kodi forum](https://forum.kodi.tv/showthread.php?tid=25522) and [GitHub repository](https://github.com/retrospect-addon/plugin.video.retrospect).

### Rules & Terms ###

> With more people adding channels for Retrospect, here are some rules and terms for channels we will host. Please ensure your channel qualifies before asking for it to be merged with the master branch:

Reviewer: Perhaps this should be more something like: "If you want to contribute by adding a new channel, feel free to start a new PR with those changes. Before doing this please ensure the new channel qualifies with the following rules:

Reviewer 2: Sorry for me commenting a little late. 

I think this would be great! To get other people than already overworked you, Bas, to handle some channels would be good.

> With more people adding channels for Retrospect, here are some rules and terms for channels we will host. Please ensure your channel qualifies before asking for it to be merged with the master branch:

Author: ```suggestion
Please ensure any channel you want to add qualifies before asking for it to be included or doing the work to make it so:
```
